src/main/java/org/icgc/argo/workflow_management/service/NextflowService.java

```diff
@@ -0,0 +1,185 @@
+    val launcherParams = new HashMap<String, Object>();
+    val cliOptions = new CliOptions();
+    cliOptions.setBackground(true);
+    launcherParams.put("options", cliOptions);
+
+    return createWithReflection(Launcher.class, launcherParams)
+        .orElseThrow(NextflowReflectionException::new);
+  }
+
+  private CmdKubeRun createCmd(@NonNull Launcher launcher, @NonNull WESRunParams params)
+      throws NextflowReflectionException {
+
+    // Config from application.yml
+    val k8sConfig = config.getK8s();
+    val webLogUrl = config.getWeblogUrl();
+
+    // params to build CmdKubeRun object
+    val cmdParams = new HashMap<String, Object>();
+
+    // always pull latest code before running
+    // (does not prevent us running a specific version (revision),
+    // does enforce pulling of that branch/hash before running)
+    cmdParams.put("latest", true);
+
+    // launcher and launcher options required by CmdKubeRun
+    cmdParams.put("launcher", launcher);
+
+    // workflow name/git and workflow params from request
+    cmdParams.put("args", Arrays.asList(params.getWorkflowUrl()));
+    cmdParams.put("params", params.getWorkflowParams());
+
+    // K8s options from application.yml
+    cmdParams.put("namespace", k8sConfig.getNamespace());
+    cmdParams.put("volMounts", Collections.singletonList(k8sConfig.getVolMounts()));
+
+    // Where to POST event-based logging
+    cmdParams.put("withWebLog", webLogUrl);
+
+    // Dynamic engine properties/config
+    val workflowEngineOptions = params.getWorkflowEngineParameters();
+
+    // Use revision if provided in workflow_engine_options
+    if (nonNull(workflowEngineOptions)) {
+
+      if (nonNull(workflowEngineOptions.getWorkflowVersion())) {
+        cmdParams.put("revision", workflowEngineOptions.getWorkflowVersion());
+      }
+
+      // Process options (default docker container to run for process if not specified)
+      if (nonNull(workflowEngineOptions.getDefaultContainer())) {
+        val processOptions = new HashMap<String, String>();
+        processOptions.put("container", workflowEngineOptions.getDefaultContainer());
+        cmdParams.put("process", processOptions);
+      }
+    }
+
+    return createWithReflection(CmdKubeRun.class, cmdParams)
+        .orElseThrow(NextflowReflectionException::new);
+  }
+
+  private K8sDriverLauncher createDriver(@NonNull CmdKubeRun cmd)
+      throws NextflowReflectionException {
+
+    Method checkRunName = null;
+
+    try {
+      checkRunName = CmdKubeRun.class.getDeclaredMethod("checkRunName");
+    } catch (NoSuchMethodException e) {
+      e.printStackTrace();
+    }
+
+    if (checkRunName != null) {
+      checkRunName.setAccessible(true);
+
+      try {
+        checkRunName.invoke(cmd);
+      } catch (IllegalAccessException | InvocationTargetException e) {
+        e.printStackTrace();
+      }
+    } else {
+      throw new NextflowReflectionException("Cannot access checkRunName!");
+    }
```

Comment:
This seems like stateless logic. Create a static method for this, that takes any object (i.e `CmdKubeRun`) and executes the input method name without any parameters (i.e `checkRunName`). So something like:
```java
public static <C extends CmdRun> void nextflowCmdVoidInvoke(C nextflowCmd, String argumentlessMethodToInvoke, Object ... args){...}
```
so you might make a call like
```java
nextflowCmdVoidInvoke(cmd, "checkRunName");
```